internal/playwright/config.go

```diff
@@ -24,34 +23,35 @@ var supportedBrwsList = []string{"chromium", "firefox", "webkit"}
 
 // Project represents the playwright project configuration.
 type Project struct {
-	config.TypeDef `yaml:",inline"`
+	config.TypeDef `yaml:",inline" mapstructure:",squash"`
 	ShowConsoleLog bool
 	ConfigFilePath string             `yaml:"-" json:"-"`
 	Sauce          config.SauceConfig `yaml:"sauce,omitempty" json:"sauce"`
 	Playwright     Playwright         `yaml:"playwright,omitempty" json:"playwright"`
-	Suites         []Suite            `yaml:"suites,omitempty" json:"suites"`
-	BeforeExec     []string           `yaml:"beforeExec,omitempty" json:"beforeExec"`
-	Docker         config.Docker      `yaml:"docker,omitempty" json:"docker"`
-	Npm            config.Npm         `yaml:"npm,omitempty" json:"npm"`
-	RootDir        string             `yaml:"rootDir,omitempty" json:"rootDir"`
-	RunnerVersion  string             `yaml:"runnerVersion,omitempty" json:"runnerVersion"`
-	Artifacts      config.Artifacts   `yaml:"artifacts,omitempty" json:"artifacts"`
-	Defaults       config.Defaults    `yaml:"defaults,omitempty" json:"defaults"`
+	// Suite is only used as a workaround to parse adhoc suites that are created via CLI args.
+	Suite         Suite             `yaml:"suite,omitempty" json:"-"`
+	Suites        []Suite           `yaml:"suites,omitempty" json:"suites"`
+	BeforeExec    []string          `yaml:"beforeExec,omitempty" json:"beforeExec"`
+	Docker        config.Docker     `yaml:"docker,omitempty" json:"docker"`
+	Npm           config.Npm        `yaml:"npm,omitempty" json:"npm"`
+	RootDir       string            `yaml:"rootDir,omitempty" json:"rootDir"`
+	RunnerVersion string            `yaml:"runnerVersion,omitempty" json:"runnerVersion"`
+	Artifacts     config.Artifacts  `yaml:"artifacts,omitempty" json:"artifacts"`
+	Defaults      config.Defaults   `yaml:"defaults,omitempty" json:"defaults"`
+	Env           map[string]string `yaml:"env,omitempty" json:"env"`
 }
 
 // Playwright represents crucial playwright configuration that is required for setting up a project.
 type Playwright struct {
-	// Deprecated. ProjectPath is succeeded by Project.RootDir.
-	ProjectPath string `yaml:"projectPath,omitempty" json:"projectPath,omitempty"`
-	Version     string `yaml:"version,omitempty" json:"version,omitempty"`
+	Version string `yaml:"version,omitempty" json:"version,omitempty"`
 }
 
 // Suite represents the playwright test suite configuration.
 type Suite struct {
 	Name              string            `yaml:"name,omitempty" json:"name"`
 	Mode              string            `yaml:"mode,omitempty" json:"-"`
 	PlaywrightVersion string            `yaml:"playwrightVersion,omitempty" json:"playwrightVersion,omitempty"`
-	TestMatch         string            `yaml:"testMatch,omitempty" json:"testMatch,omitempty"`
+	TestMatch         []string          `yaml:"testMatch,omitempty" json:"testMatch,omitempty"`
```

Comment:
Just fyi that this is backwards compatible thanks to viper (or its parser). Folio and playwright-test can both worth with arrays.
And viper's parser is capable of mapping string values to arrays (not the other way around ofc, since you'd have an information loss).